### PR TITLE
[CCAP-359][CCAP-437] Handle application already submitted, responded to, and/or invalid confirmation code on providerresponse/application-id screen

### PR DESCRIPTION
**Pull request by cram-cfa**
#### 🔗 Jira ticket
https://codeforamerica.atlassian.net/browse/CCAP-359
https://codeforamerica.atlassian.net/browse/CCAP-437

#### ✅ Completion tasks

- [ ] Added relevant tests
- [ ] Meets acceptance criteria

### CODE REVIEW

File: src/main/java/org/ilgcc/app/submission/actions/CheckClientSubmissionForProvider.java [hunk 1]
```diff
@@ -46,7 +46,14 @@ public CheckClientSubmissionForProvider(HttpSession httpSession, MessageSource m
 
     @Override
     public void run(Submission submission) {
+
         UUID clientSubmissionId = (UUID) httpSession.getAttribute(SESSION_KEY_CLIENT_SUBMISSION_ID);
+        String providerSubmissionStatus = (String) httpSession.getAttribute(SESSION_KEY_CLIENT_SUBMISSION_STATUS);
+
+        if (clientSubmissionId != null && providerSubmissionStatus != null) {
```
Comment (analoo) [inline src/main/java/org/ilgcc/app/submission/actions/CheckClientSubmissionForProvider.java:53]: what happens if the providerSubmissionStatus is set to active.

Comment (analoo) [reply src/main/java/org/ilgcc/app/submission/actions/CheckClientSubmissionForProvider.java:53]: So if you come into submit-start from a link, status should be null however if we come to submit-start after the familyapplicationcode screen then the status would already be set.

The status could be set to active but in that case, we would not get to the submit-start screen. So this makes sense.

Comment (cram-cfa) [reply src/main/java/org/ilgcc/app/submission/actions/CheckClientSubmissionForProvider.java:53]: yep! that's exactly correct-- when you come from a link, everything is null and the status gets set to Active/Expired/Responded. But if you're coming without the link, you should only wind up back here if the status is Expired/Responded... and Active continues in the flow.

**Comment by cram-cfa** [comment]
> This makes sense to me. I didn't test the logic.
> 
> Would you mind adding tests of these changes to just validate that it works as expected?
> 
> https://github.com/codeforamerica/il-gcc-form-flow/blob/94c4441d806d6c753e4aa54f72055578a347fb17/src/test/java/org/ilgcc/app/journeys/ProviderresponseFlowJourneyTest.java#L13

Yep, I'll add some tests in another PR so we can merge this in and unblock / confirm the ticket Alex was about to work on!

**Pull request merged**